Xwt.XamMac/Xwt.Mac/WindowFrameBackend.cs

```diff
@@ -334,20 +331,22 @@ protected virtual void OnBoundsChanged ()
 
 		void IWindowFrameBackend.SetTransientFor (IWindowFrameBackend parent)
 		{
+			//TODO: why this?
 			if (!((IWindowFrameBackend)this).ShowInTaskbar)
 				Window.StyleMask &= ~NSWindowStyle.Miniaturizable;
 
-			var win = Window as NSWindow ?? ApplicationContext.Toolkit.GetNativeWindow (parent) as NSWindow;
-
-			if (Window.ParentWindow != win) {
+			//we try to get the native object from the parameter if not we fallback into the real parent
+			if (ApplicationContext.Toolkit.GetNativeWindow(parent) is NSWindow nParent && nParent != Window.ParentWindow)
+            {
 				// remove from the previous parent
 				if (Window.ParentWindow != null)
-					Window.ParentWindow.RemoveChildWindow (Window);
+					Window.ParentWindow.RemoveChildWindow(Window);
+
+				// new parent has any associed window we remove it
+				if (nParent.ParentWindow != null)
+					nParent.ParentWindow.RemoveChildWindow(nParent);
```

Comment:
Is this needed? Does it matter if the nParent has a parent window? We are making Window have nParent as its ParentWindow. What nParent already has a parent should not matter? But maybe I am not understanding.

I think we only need `Window.ParentWindow.RemoveChildWindow(Window);` so this window does not have the old parent window.

Reply:
fixed